include/unix/osd.h

```diff
@@ -37,6 +37,11 @@
 #include <unistd.h>
 #include <errno.h>
 
+#define OFI_WAIT_MUTEX(mutex_cond) ((mutex_cond)->mutex)
+#define OFI_WAIT_COND(mutex_cond)  ((mutex_cond)->cond)
+
+#define FI_WAIT_MUTEX_CRITSEC FI_WAIT_MUTEX_COND
```

Comment:
You're still wanting to abstract mutexes and critical sections.  Native wait objects should not be abstracted -- that's the benefit and disadvantage to using them.  The code that handles critical sections should be separate from the code that processes mutexes.  They call different functions and work on different platforms.  Please remove anything that attempts to abstract the differences.